WordPress/wordcamp.org
# Blocks: Use `@wordpress/scripts` for builds

Use `wp-scripts` commands to run builds and linting, and update the webpack config to merge in default WP configuration. This enables the `DependencyExtractionWebpackPlugin` plugin, so that `@wordpress/*` scripts are automatically extracted into a `*.deps.json` file, which we can read to automatically generate the correct dependencies for enqueuing.

To get the benefit of dependency extraction, we need to import the wordpress components as `import`s,  rather than const declarations - the wp-scripts webpack also takes care of converting those to the `wp.` globals, so we're not adding any extra libraries.

Post switch, file size effectively doesn't change:

- **blocks.min.js:**  before 163 KiB; after 163 KiB
- **blocks.min.css:**  before 3.82 KiB; after 3.92 KiB

It does add the `blocks.min.deps.json` file, but this is only read server-side.

The switch also brought in slightly updated eslint & stylelint config, so there were a few new issues. This PR also fixes those issues:

- in `ImageSizeControl` & `ImageAlignmentControl`, the BaseControl is used with label but without an id – this flagged [the `no-base-control-with-label-without-id`,](https://github.com/WordPress/gutenberg/blob/master/packages/eslint-plugin/docs/rules/no-base-control-with-label-without-id.md) because without an ID the label isn't attached to anything. Turns out we don't want to use a literal `<label>` in these contexts anyway, so I've moved the label to in-component `span`s.
- in the scss files, remove the unit from `line-height` and remove a redundant `&`.

Fixes #71

**To test**

- Run `npm run build` or `npm start` to build the files
- Check that each block still works
- Run `npm run lint:js`
- Run `npm run lint:css`
- There should be no errors
